shared/chat/conversation/input-area/normal/platform-input.native.tsx

```diff
@@ -211,39 +212,94 @@ class _PlatformInput extends PureComponent<PlatformInputPropsInternal, State> {
 }
 const PlatformInput = AddSuggestors(_PlatformInput)
 
-const Action = ({hasText, insertMentionMarker, isEditing, onSubmit, openFilePicker, openMoreMenu}) =>
-  hasText ? (
-    <Kb.Button
-      type="Default"
-      small={true}
-      style={styles.send}
-      onClick={onSubmit}
-      label={isEditing ? 'Save' : 'Send'}
-    />
-  ) : (
-    <Kb.Box2 direction="horizontal" style={styles.actionIconsContainer}>
-      <Kb.Icon
-        onClick={insertMentionMarker}
-        type="iconfont-mention"
-        style={Kb.iconCastPlatformStyles(styles.actionButton)}
-        fontSize={22}
-      />
-      {smallGap}
-      <Kb.Icon
-        onClick={openFilePicker}
-        type="iconfont-camera"
-        style={Kb.iconCastPlatformStyles(styles.actionButton)}
-        fontSize={22}
-      />
-      {smallGap}
-      <Kb.Icon
-        onClick={openMoreMenu}
-        type="iconfont-add"
-        style={Kb.iconCastPlatformStyles(styles.actionButton)}
-        fontSize={22}
-      />
-    </Kb.Box2>
-  )
+type ActionProps = {
+  hasText: boolean
+  onSubmit: () => void
+  isEditing: boolean
+  openFilePicker: () => void
+  openMoreMenu: () => void
+  insertMentionMarker: () => void
+}
+
+const Action = React.memo(
+  ({hasText, insertMentionMarker, isEditing, onSubmit, openFilePicker, openMoreMenu}: ActionProps) => {
+    const hasValue = React.useRef(new Kb.NativeAnimated.Value(hasText ? 1 : 0)).current
+    const prevHasText = Container.usePrevious(hasText)
+
+    React.useEffect(() => {
+      Kb.NativeAnimated.timing(hasValue, {
+        duration: 200,
+        toValue: hasText ? 1 : 0,
+        useNativeDriver: true,
+      }).start()
+    }, [hasText, prevHasText, hasValue])
```

Comment:
do we want to call `stop()` in cleanup?

Reply:
Also why does the `prevHasText` matter here? It's just the the value we are changing into that matters right?

Reply:
fixed

Reply:
i don't think we need to call stop, i think its benign to let it run